### PR TITLE
Br merchants find all

### LINKED ISSUE #3: Record Endpoint - merchants find
* always case insensitive
* returns one single object representation

type:
GET /api/v1/merchants/find?parameters

examples:
GET /api/v1/merchants/find?id=1
GET /api/v1/merchants/find?name=Schroeder-Jerde
GET /api/v1/merchants/find?created_at=date
GET /api/v1/merchants/find?updated_at=date

### PR DESCRIPTION
**Pull request by riverswb**
@lucyconklin @case-eee

closes #3 #4 

Updates the merchants table to use citext and timestamps to have a precision:0
Adds Merchant Find Controller to find a single merchant by; id, name (case insensitive), created at, updated at
Adds Merchant find_all and test for each
Adds Merchant random controller, route, and test
All tests passing

### CODE REVIEW

File: app/controllers/api/v1/merchants/find_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+  def show
+    render json: Merchant.find_by(find_params)
+  end
+
+  def index
+    render json: Merchant.where(find_params)
+  end
+private
+
```
Comment (lucyconklin) [inline app/controllers/api/v1/merchants/find_controller.rb:11]: I think hound was yelling at us on the last project for not indenting "private".

File: app/controllers/api/v1/merchants/random_controller.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Api::V1::Merchants::RandomController < ApplicationController
+  def show
+    render json: Merchant.order("RANDOM()").first
+  end
```
Comment (lucyconklin) [inline app/controllers/api/v1/merchants/random_controller.rb:4]: such a great find!

File: spec/requests/api/v1/merchants/merchants_find_all_requests_spec.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    create_list(:merchant, 5)
+
+    get '/api/v1/merchants/random'
+
+    found = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(found).to be_a(Hash)
+    expect(found["id"]).to be_between(1,5).inclusive
+  end
```
Comment (lucyconklin) [inline spec/requests/api/v1/merchants/merchants_find_all_requests_spec.rb:60]: Oh good call, I think I will add this line to my random test (59)

**Comment by lucyconklin** [comment]
Looks great! I am going to merge this into my current branch and keep working.

File: app/controllers/api/v1/merchants/find_controller.rb (hunk 1 above)
Comment (case-eee) [reply app/controllers/api/v1/merchants/find_controller.rb:11]: yup - indent this!

File: app/controllers/api/v1/merchants/random_controller.rb (hunk 2 above)
Comment (case-eee) [reply app/controllers/api/v1/merchants/random_controller.rb:4]: @riverswb definitely a great find! let's move this method out to the model though. maybe a method in our `Merchant` model that's called `random`?

**Review by case-eee — COMMENTED** [review]
👍

**Pull request merged**